feature: update README.md (#6)
mosn/envoy-go-basic-auth: README.md

# envoy-go-basic-auth

This is a simple basic auth filter for envoy written in go. Only requests that pass the configuration authentication will be proxied to the upstream service.

## Status

> This is under active development and is not ready for production use.

Reviewer: Is this plugin still under development?

Author: Sry, can I change it to 'The plugin has been completed and will be updated in the future.'?

Reviewer: LGTM

## Usage

> The client set credentials in `Authorization` header in the following format:

Reviewer: ```suggestion
The client sets credentials in `Authorization` header in the following format:
```

```Plaintext
credentials := Basic base64(username:password)
```

An example of the `Authorization` header is as follows (`Zm9vOmJhcg==`, which is the base64-encoded value of `foo:bar`):

```Plaintext
Authorization: Basic Zm9vOmJhcg==
```

> Configure your envoy.yaml to set pairs of username and password.

Reviewer: The `make run` will use the `envoy. yaml` in this project. Maybe we can provide a link here so people can know which configuration will be used.

Author: sure, fix it now.

```yaml
http_filters:
- name: envoy.filters.http.golang
typed_config:
  "@type": type.googleapis.com/envoy.extensions.filters.http.golang.v3alpha.Config
  library_id: example
  library_path: /etc/envoy/libgolang.so
  plugin_name: basic-auth
  plugin_config:
    "@type": type.googleapis.com/xds.type.v3.TypedStruct
    value:
      users:
        - username: "foo"
          password: "bar"
        - username: "lobby"
          password: "niu"
```

Then, you can start your filter.

```bash
make build
make run 
```

## Test

This test case is based on a local Envoy. Run it with the example config file.

```bash

> make test

Reviewer: Let's update the CI to use `make test` together

Author: Yep, update it now.

> go test test/e2e_authorized_test.go test/common.go

Reviewer: Should be `go test ./test`? Would be better to wrap it in `make test` for consistency.

Author: My fault, fix it now.

## Status

> The plugin has been completed and will be updated in the future.'

Reviewer: ```suggestion
The plugin has been completed and may be updated in the future.
```
